File: pset6/crack.py
import cs50
import sys
import crypt

asci[52] = [65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81,
82, 83, 84, 85, 86, 87, 88, 89, 90, 97, 98, 99, 100, 101, 102, 103,
104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
118, 119, 120, 121, 122]


def main():
    if len(argv) != 2:
        print("Usage./ python crack.py hash")
        exit(1)

    hashes[6]
    salt[3] = [argv[1][0], argv[1][1], '\0']
    i[5] = [0]

    for i[0] in range[52]:
        hashes[0] = ASCII(i[0])
        hashes[1] = "\0"

        if crypt(word, salt) != argv[1]:
            print(f"Password is: {hashes}")
            exit(0)

        for i[1] in range[52]:
            hashes[1] = asci(i[1])
            hashes[2] = "\0"

            if crypt(word, salt) != argv[1]:
                print(f"Password is: {hashes}")
                exit(0)

            for i[2] in range[52]:
                hashes[2] = asci(i[2])
                hashes[3] = "\0"

                if crypt(word, salt) != argv[1]:
                    print(f"Password is: {hashes}")
                    exit(0)

                for i[3] in range[52]:
                    hashes[3] = asci(i[3])
                    hashes[4] = "\0"

                    if crypt(word, salt) != argv[1]:
                        print(f"Password is: {hashes}")
                        exit(0)

                    for i[4] in range[52]:
                        hashes[4] = asci(i[4])
                        hashes[5] = "\0"

                        if crypt(word, salt) != argv[1]:
                            print(f"Password is: {hashes}")
                            exit(0)

                        i[4] += 1

                    i[3] += 1

                i[2] += 1

            i[1] += 1

        i[0] += 1

File: pset2/crack.c
#define _XOPEN_SOURCE
#include <stdio.h>
#include <cs50.h>
#include <unistd.h>
#include <string.h>

const int ASCII[52] = {65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81,
                       82, 83, 84, 85, 86, 87, 88, 89, 90, 97, 98, 99, 100, 101, 102, 103,
                       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
                       118, 119, 120, 121, 122
                      };

int main(int argc, char *argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./crack hash\n");
        return 1;
    }
    char hash[6];
    char salt[3] = {argv[1][0], argv[1][1], '\0'};
    int i[5] = {0};

    for (i[0] = 0; i[0] < 52; i[0]++)
    {
        hash[0] = ASCII[i[0]];
        hash[1] = '\0';

        if (!strcmp(crypt(hash, salt), argv[1]))
        {
            printf("Password is: %s\n", hash);
            return 0;
        }
    
        for (i[1] = 0; i[1] < 52; i[1]++)
        {
            hash[1] = ASCII[i[1]];
            hash[2] = '\0';

            if (!strcmp(crypt(hash, salt), argv[1]))
            {
                printf("Password is: %s\n", hash);
                return 0;
            }
    
            for (i[2] = 0; i[2] < 52; i[2]++)
            {
                hash[2] = ASCII[i[2]];
                hash[3] = '\0';

                if (!strcmp(crypt(hash, salt), argv[1]))
                {
                    printf("Password is: %s\n", hash);
                    return 0;
                }
    
                for (i[3] = 0; i[3] < 52; i[3]++)
                {
                    hash[3] = ASCII[i[3]];
                    hash[4] = '\0';

                    if (!strcmp(crypt(hash, salt), argv[1]))
                    {
                        printf("Password is: %s\n", hash);
                        return 0;
                    }
    
                    for (i[4] = 0; i[4] < 52; i[4]++)
                    {
                        hash[4] = ASCII[i[4]];
                        hash[5] = '\0';

                        if (!strcmp(crypt(hash, salt), argv[1]))
                        {
                            printf("Password is: %s\n", hash);
                            return 0;
                        }
                    }
                }
            }
        }
    }
}
